samza-yarn/src/main/scala/org/apache/samza/job/yarn/ClientHelper.scala

```diff
@@ -249,8 +249,23 @@ class ClientHelper(conf: Configuration) extends Logging {
         .toList
   }
 
+  def getPreviousApplicationIds(appName: String): List[ApplicationId] = {
+    val getAppsRsp = yarnClient.getApplications
+
+    getAppsRsp
+      .asScala
+      .filter(appRep => ( !(
```

Comment:
Minor: Maybe be cleaner to write this condition as a multi-line block using `{}` with an extracted val for appStatus.
Also, not clear what Rsp and Rep are. Maybe use a more verbose name?